receiver/mongodbatlasreceiver/access_logs.go

```diff
@@ -0,0 +1,407 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package mongodbatlasreceiver // import "github.com/open-telemetry/opentelemetry-collector-contrib/receiver/mongodbatlasreceiver"
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"sync"
+	"time"
+
+	"go.mongodb.org/atlas/mongodbatlas"
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/consumer"
+	"go.opentelemetry.io/collector/extension/experimental/storage"
+	"go.opentelemetry.io/collector/pdata/pcommon"
+	"go.opentelemetry.io/collector/pdata/plog"
+	rcvr "go.opentelemetry.io/collector/receiver"
+	"go.uber.org/zap"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/receiver/mongodbatlasreceiver/internal"
+)
+
+const (
+	accessLogStorageKey           = "last_endtime_access_logs"
+	defaultAccessLogsPollInterval = time.Minute
+	defaultAccessLogsPageSize     = 20000
+	defaultAccessLogsMaxPages     = 10
+)
+
+type accessLogStorageRecord struct {
+	GroupID           string    `json:"group_id"`
+	ClusterName       string    `json:"cluster_name"`
+	NextPollStartTime time.Time `json:"next_poll_start_time"`
+}
+
+type accessLogClient interface {
+	GetProject(ctx context.Context, groupID string) (*mongodbatlas.Project, error)
+	GetClusters(ctx context.Context, groupID string) ([]mongodbatlas.Cluster, error)
+	GetAccessLogs(ctx context.Context, groupID string, clusterName string, opts *internal.GetAccessLogsOptions) (ret []*mongodbatlas.AccessLogs, err error)
+}
+
+type accessLogsReceiver struct {
+	client        accessLogClient
+	logger        *zap.Logger
+	storageClient storage.Client
+	cfg           *Config
+	consumer      consumer.Logs
+
+	record       []*accessLogStorageRecord
+	pollInterval time.Duration
+	authResult   *bool
+	wg           *sync.WaitGroup
+	cancel       context.CancelFunc
+}
+
+func newAccessLogsReceiver(settings rcvr.CreateSettings, cfg *Config, consumer consumer.Logs) *accessLogsReceiver {
+	r := &accessLogsReceiver{
+		cancel:        func() {},
+		client:        internal.NewMongoDBAtlasClient(cfg.PublicKey, cfg.PrivateKey, cfg.RetrySettings, settings.Logger),
+		cfg:           cfg,
+		logger:        settings.Logger,
+		consumer:      consumer,
+		pollInterval:  cfg.CollectionInterval,
```

Comment:
I feel like this is the wrong collection interval as described in the README.md this is the 

> - `collection_interval` (default `3m`) This receiver collects metrics on an interval. Valid time units are `ns`, `us` (or `µs`), `ms`, `s`, `m`, `h`.

I think there should be something in the configuration that is separate lever for users to use to determine how often we poll.

Reply:
Currently access logs are enabled per-project, so we would then have a lever pre-project to define poll interval, unless you had a different idea.

Reply:
Yeah my biggest gripe was that the documentation didn't match the usage. Could we perhaps have a field per `LogsProjectConfig` with a reasonable default? That's at least where my head was at with how I expected this to work.

Reply:
Refactored to allow poll interval config per project.